services/web/client/source/class/qxapp/dev/fake/Data.js

```diff
@@ -19,9 +19,282 @@ qx.Class.define("qxapp.dev.fake.Data", {
+              }
+            },
+            inSb: {
+              displayOrder: 5,
+              label: "Widget SelectBox Test",
+              description: "Test Input for SelectBox",
+              defaultValue: "dog",
+              type: "string",
+              widget: {
+                type: "SelectBox",
+                structure: [
+                  {
+                    key: "dog",
+                    label: "A Dog"
+                  },
+                  {
+                    key: "cat",
+                    label: "A Cat"
+                  }
+                ]
+              }
+            },
+            inFile: {
+              displayOrder: 6,
+              label: "File",
+              description: "Test Input File",
+              type: "data:*/*"
+            },
+            inImage: {
+              displayOrder: 7,
+              label: "Image",
+              description: "Test Input Image",
+              type: "data:[image/jpeg,image/png]"
+            }
+          },
+          outputs: {
+            outNumber: {
+              label: "Number Test",
+              description: "Test Output for Number",
+              displayOrder: 0,
+              type: "number"
+            },
+            outInteger: {
+              label: "Integer Test",
+              description: "Test Output for Integer",
+              displayOrder: 1,
+              type: "integer"
+            },
+            outBool: {
+              label: "Boolean Test",
+              description: "Test Output for Boolean",
+              displayOrder: 2,
+              type: "boolean"
+            },
+            outPng: {
+              label: "Png Test",
+              description: "Test Output for PNG Image",
+              displayOrder: 3,
+              type: "data:image/png"
+            }
+          }
+        }
+      };
+    },
+    getProjectList: function() {
+      return [
+        {
+          name: "Sample Project",
+          description: "A little fake project without actual backend",
+          notes: "# title\nThere be dragons inside",
+          owner: "UUID-OF-TOBI",
+          collaborators: {
+            "UUID-OF-PEDRO": [
+              "read",
+              "write"
+            ]
+          },
+          creationDate: "2018-07-02T16:01:00Z",
+          lastChangeDate: "2018-07-02T16:02:22Z",
+          workbench: {
+            "UUID1": {
+              key: "service/dynamic/itis/file-picker",
+              version: "0.0.0",
+              outputs: {
+                outFile: {
+                  store: "s3-z43",
+                  path: "/bucket1/file1"
+                }
+              },
+              position: {
+                x: 10,
+                y: 10
+              }
+            },
+            "UUID2": {
+              key: "service/computational/itis/sleeper",
+              version: "0.0.0",
+              inputs: {
+                inNumber: 3.5
+              },
+              outputs: {
```

Comment:
Are all the inputs/outputs sub entries now optional?

Reply:
at the moment there is no provision of defining mandatory and optional inputs ... if there was it would have to be added to the node schema ... this here is just the payload